app/script/user/UserService.js

```diff
@@ -0,0 +1,290 @@
+   * @returns {Promise} Promise that will resolve with the self user
+   */
+  get_own_user() {
+    return this.client.send_request({
+      type: 'GET',
+      url: this.client.create_url(UserService.URL.SELF),
+    });
+  }
+
+  /**
+   * Update your own user profile.
+   * @option data {Array<z.assets.Asset>} picture
+   * @option data {number} accent_id
+   * @option data {String} name
+   * @see https://staging-nginz-https.zinfra.io/swagger-ui/#!/users/updateSelf
+   * @param {Object} data - Updated user profile information
+   * @returns {Promise} Resolves with backend response.
+   */
+  update_own_user_profile(data) {
+    return this.client.send_json({
+      data,
+      type: 'PUT',
+      url: this.client.create_url(UserService.URL.SELF),
+    });
+  }
+
+  /**
+   * Change your own user email.
+   * @see https://staging-nginz-https.zinfra.io/swagger-ui/#!/users/changeEmail
+   * @param {string} email - New email address for the user
+   * @returns {Promise} Promise that resolves when email changing process has been started on backend
+   */
+  change_own_email(email) {
+    return this.client.send_json({
+      data: {
+        email,
+      },
+      type: 'PUT',
+      url: this.client.create_url('/self/email'),
+    });
+  }
+
+  /**
+   * Change username.
+   * @param {string} username - New username for the user
+   * @returns {Promise} Promise that resolves when username changing process has been started on backend
+   */
+  change_own_username(username) {
+    return this.client.send_json({
+      data: {
+        handle: username,
+      },
+      type: 'PUT',
+      url: this.client.create_url('/self/handle'),
+    });
+  }
+
+  /**
+   * Change own user password.
+   * @see https://staging-nginz-https.zinfra.io/swagger-ui/#!/users/changePassword
+   * @param {string} new_password - New user password
+   * @param {string} old_password - Old password of the user (optional)
+   * @returns {Promise} Promise that resolves when password has been changed on backend
+   */
+  change_own_password(new_password, old_password) {
+    return this.client.send_json({
+      data: {
+        new_password,
+        old_password,
+      },
+      type: 'PUT',
+      url: this.client.create_url('/self/password'),
+    });
+  }
+
+  /**
+   * Change your phone number.
+   * @see https://staging-nginz-https.zinfra.io/swagger-ui/#!/users/changePhone
+   * @param {string} phone_number - Phone number in E.164 format
+   * @returns {Promise} Promise that resolves when phone number change process has been started on backend
+   */
+  change_own_phone_number(phone_number) {
+    return this.client.send_json({
+      data: {
+        phone: phone_number,
+      },
+      type: 'PUT',
+      url: this.client.create_url('/self/phone'),
+    });
+  }
+
+  /**
+   * Delete self user.
+   * @returns {Promise} Promise that resolves when account deletion has been initiated
+   */
+  delete_self() {
+    return this.client.send_json({
+      data: {
```

Comment:
Has anyone ever checked why we send this data content? Seems like an obsolete placeholder that should be removed.